dragonflybsd/esrt-api.md

```diff
@@ -0,0 +1,92 @@
+# Instructions for getting user-space ESRT API on DragonFlyBSD
+
+Read [README.md](./README.md) first for general information.
+
+The plan is:
+ * obtain `ioctl-efitool` branch of https://github.com/3mdeb/DragonFlyBSD/tree/esrt
+ fork of DragonFlyBSD source tree
+ * compile kernel
+ * compile EFI bootloader
+ * install both of them
+ * compile and install tool that uses ESRT API
+ * reboot and check that it works
+
+## Preparation
+
+As a root on installed DragonFlyBSD:
+
+```
+# cd /usr
+# make src-create-shallow
+# make src-update
+```
+
+Sources initialization may take a while. After that, add 3mdeb repository:
+
+```
+# git remote add 3mdeb https://github.com/3mdeb/DragonFlyBSD.git
+# git fetch 3mdeb
+# git checkout 3mdeb/ioctl-efitool
+```
+
+## Kernel compilation and installation
+
+```
+# cd /usr/src/sys/config
+# mkdir /root/kernels
+# cp X86_64_GENERIC /root/kernels/MYKERNEL
+# ln -s /root/kernels/MYKERNEL
+
+# make -j4 nativekernel KERNCONF=MYKERNEL
+# make installkernel KERNCONF=MYKERNEL
+```
+
+## EFI bootloader compilation and installation
+
+```
+# cd /usr/src/stand/boot/efi/libefi
+# make
+
+# cd /usr/src/stand/boot/efi/loader
+# make
```

Comment:
@kotylamichal this returns lots of undefined references

Reply:
@kotylamichal needed to go into `cd /usr/src/stand` run make there (which returns errors) and then could compile loader

Reply:
This should be fixed, maybe we just need to compile something more before the loader?

Reply:
@kotylamichal worked without rebuilding bootloader so we can get rid off this section

Reply:
removed